scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java

```diff
@@ -745,6 +745,7 @@ public synchronized void prepareTasks() {
         tasks.clear();
         // re-init taskId
         int id = 0;
+        this.jobInfo.setVariables(this.getVariablesAsReplacementMap());
```

Comment:
Very good find ! It all comes to the fact that the jobInfo is not initialized when the job is submitted (created using the empty no arg constructor of JobInfoImpl). The best here would be to properly initialize it, which is partially done in jobSubmitted.submitAction().